kore/src/Kore/Attribute/Pattern/ConstructorLike.hs

```diff
@@ -95,6 +95,11 @@ instance Synthetic ConstructorLike (Application Symbol) where
         , childrenAreConstructorLike =
             ConstructorLike . Just $ ConstructorLikeHead
 
+        | Symbol.isConstructor symbol
+        , childrenAreConstructorLike
+        , childrenAreNotSortInjections =
```

Comment:
When the constructor is overloaded, the children are _expected_ to be sort injections.